tests/test_awslambda/utilities.py

```diff
@@ -166,14 +166,18 @@ def create_invalid_lambda(role):
 def get_role_name():
     with mock_iam():
         iam = boto3.client("iam", region_name=_lambda_region)
-        try:
-            return iam.get_role(RoleName="my-role")["Role"]["Arn"]
-        except ClientError:
-            return iam.create_role(
-                RoleName="my-role",
-                AssumeRolePolicyDocument="some policy",
-                Path="/my-path/",
-            )["Role"]["Arn"]
+        while True:
+            try:
+                return iam.get_role(RoleName="my-role")["Role"]["Arn"]
```

Comment:
So, I don't really remember why I chose to have a single role, but I wonder if it would be easier to just create a new role each time, with a unique name?
AFAIK there is no actual need to share a single role across all functions - creating a new role, one per function/test, may be a lot cleaner and less error-prone.

Reply:
Unfortunately, this helper method is called ~100 times throughout the test suite and numerous tests make assertions based on the `my-role` name.  Additionally, the method returns the role's ARN, which would require some parsing to extract the random name (which, again, is currently assumed to be `my-role` in many tests).

I'm with you that this should ultimately be simplified/refactored, but I don't know if I feel like doing the work at the moment.  The changes in this PR at least have the advantage of only being run as part of the tests (and not moto itself).  I can't stand flaky CI (especially when my PRs end up with a red x as a result 😄 ), but I leave it up to you as to whether or not you think that's worth merging it as is or not.